model/map/src/main/java/org/keycloak/models/map/realm/MapRealmProvider.java

```diff
@@ -68,7 +76,9 @@ public RealmModel createRealm(String id, String name) {
             throw new ModelDuplicateException("Realm with given name exists: " + name);
         }
 
-        if (id != null && tx.read(id) != null) {
+        id = UuidValidator.validateAndConvert(id, "realm", Boolean.valueOf(
```

Comment:
The JavaDoc states "Internal ID of the realm or null if one is to be created by the underlying store" for the id attribute. Please consider changing the JavaDoc to indicate that the storage might overwrite/change the ID at its own discretion.

Reply:
Updated the javadoc.

Reply:
+1 for the javadoc. Actually I don't agree with the throwing an exception, it should always create a new one.

Reply:
Based on the other comments I'll remove the exception part.

Reply:
Done, javadoc updated.